modules/web-console/src/main/java/org/apache/ignite/console/WebConsoleServer.java

```diff
@@ -309,61 +293,61 @@ private void sendResult(RoutingContext ctx, JsonObject data) {
      * @param e Error to send.
      */
     private void sendError(RoutingContext ctx, String msg, Throwable e) {
-        logError(msg, e);
+        ignite.log().error(msg, e);
 
         sendStatus(ctx, HTTP_INTERNAL_ERROR, msg + ": " + errorMessage(e));
     }
 
     /**
-     * Get the authenticated user (if any).
-     * If user not found, send {@link HttpURLConnection#HTTP_UNAUTHORIZED}.
-     *
      * @param ctx Context
-     * @return User or {@code null} if the current user is not authenticated.
      */
-    @Nullable private User checkUser(RoutingContext ctx) {
-        User user = ctx.user();
-
-        if (user == null)
-            sendStatus(ctx, HTTP_UNAUTHORIZED);
+    private void getAccount(RoutingContext ctx) {
+        try {
+            User user = ctx.user();
 
-        return user;
-    }
+            if (user == null) {
+                sendStatus(ctx, HTTP_UNAUTHORIZED);
 
-    /**
-     * @param ctx Context
-     */
-    private void handleUser(RoutingContext ctx) {
-        User user = checkUser(ctx);
+                return;
+            }
 
-        if (user != null)
             sendResult(ctx, user.principal());
+        }
+        catch (Throwable e) {
+            sendStatus(ctx, HTTP_UNAUTHORIZED);
+        }
     }
 
     /**
      * @param ctx Context
      */
-    private void handleSignUp(RoutingContext ctx) {
-        JsonObject authInfo = ctx.getBodyAsJson();
+    private void registerAccount(RoutingContext ctx) {
+        try {
+            JsonObject entries = ctx.getBodyAsJson();
 
-        authInfo.put("signup", true);
+            if (entries.getBoolean("user.admin", false)) {
```

Comment:
Looks like a bug. JosnObject does not have support for "a.x.y" properties access.